src/hffs/fs.py

```diff
@@ -118,35 +89,76 @@ def _resolve_repo_id(self, path: str) -> Tuple[str, str, str]:
             repo_type = huggingface_hub.constants.REPO_TYPES_MAPPING[repo_type]
         else:
             repo_type = huggingface_hub.constants.REPO_TYPE_MODEL
+        revision = None
         if path.count("/") > 0:
-            repo_id_with_namespace = "/".join(path.split("/")[:2])
-            path_in_repo_with_namespace = "/".join(path.split("/")[2:])
-            repo_id_without_namespace = path.split("/")[0]
-            path_in_repo_without_namespace = "/".join(path.split("/")[1:])
-            if self._repo_exists(repo_id_with_namespace, repo_type=repo_type):
-                repo_id = repo_id_with_namespace
-                path_in_repo = path_in_repo_with_namespace
-            elif self._repo_exists(repo_id_without_namespace, repo_type=repo_type):
-                repo_id = repo_id_without_namespace
-                path_in_repo = path_in_repo_without_namespace
+            if "@" in path:
+                repo_id, revision = path.split("@", 1)
+                if "/" in revision:
+                    revision, path_in_repo = revision.split("/", 1)
+                else:
+                    path_in_repo = ""
+                revision = unquote(revision)
+                if not self._repo_exists(repo_id, repo_type=repo_type):
+                    raise FileNotFoundError(f"No such repository: '{repo_id}'")
+                return repo_type, repo_id, revision, path_in_repo
             else:
-                raise FileNotFoundError(f"No such repository: '{repo_id_with_namespace}'")
+                revision = None
+                repo_id_with_namespace = "/".join(path.split("/")[:2])
+                path_in_repo_with_namespace = "/".join(path.split("/")[2:])
+                repo_id_without_namespace = path.split("/")[0]
+                path_in_repo_without_namespace = "/".join(path.split("/")[1:])
+                if self._repo_exists(repo_id_with_namespace, repo_type=repo_type):
```

Comment:
Might be worth to check if the repo exists at this exact revision. If the repo exists but not the revision, you'll get a `RevisionNotFoundError` so that you can adapt the error message below.